api-server/src/summarization/summarization.service.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  HttpException,
+  Inject,
+  Injectable,
+  LoggerService,
+} from '@nestjs/common';
+import Anthropic from '@anthropic-ai/sdk';
+import { isNullOrUndefined } from '../utils';
+import { isEmptyString } from '../utils/is-not-empty-string';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+
+@Injectable()
+export class SummarizationService {
+  private readonly client: Anthropic;
+
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER)
+    private readonly logger: LoggerService,
+  ) {
+    this.logger = logger;
+
+    if (isEmptyString(process.env['ANTHROPIC_API_KEY'])) {
+      this.logger.warn(
+        'ANTHROPIC_API_KEY is not set, the server will not be able to recommended tags',
+      );
+    } else {
+      this.client = new Anthropic({ apiKey: process.env['ANTHROPIC_API_KEY'] });
+    }
+  }
+  async getTagsFromSummary(description: string): Promise<string[]> {
+    if (isNullOrUndefined(this.client)) {
+      return [];
+    }
+
+    const sanitizedDescription = this.sanitizeDescription(description);
+
+    const prompt = `
+    Generate a set of concise, relevant tags for categorizing the following event in a database.
+    The tags should be single words or short hyphenated phrases. Focus on the event type, musical genre,
+    and key characteristics. When applicable, use one of these tags
+    'gallery, music, theater, dance, film, literary-arts, talk, festival, comedy'
+    
+    This is important, generate a valid array JSON strings.
+    
+    Example1 : [ "gallery", "music" ]
+    
+    Output JSON only.
+    Do not wrap JSON output in anything.
+    Do not try to explain the code you generated.
+    Do not add any other text to your JSON output.
+    
+    ${sanitizedDescription}
+
+    Output the tags as a JSON array of strings. Do not include any additional text or formatting."""
+    `;
+
+    const errors = [];
+    const numTries = 3;
+    for (let i = 0; i < numTries; i++) {
+      try {
+        if (numTries > 0) {
+          console.info('trying to get a valid message from anthropic again');
```

Comment:
Do these logging statements need to use `this.logger`?

Reply:
good catch, fixed